### PR TITLE
Refactoring error handling

### PR DESCRIPTION
**Pull request by y0urself**
Did some refactoring in the gmp classes:
* Unified the use of the `RequiredArgument` and `InvalidArgument`
* Introduced `InvalidArgumentType`

Added some more Error Types: `GvmServerError`, `GvmClientError`, `GvmResponseError` and `InvalidArgumentType`

* Refactored the status response errors in case of a failure in the communication with the server in `transform.py`

**Checklist**:

- [x] Tests
- [x] [CHANGELOG](https://github.com/greenbone/python-gvm/blob/master/CHANGELOG.md) Entry
- [ ] Documentation

### CODE REVIEW

File: gvm/errors.py [hunk 1]
```diff
@@ -29,6 +29,46 @@ class GvmError(Exception):
     """
 
 
+class GvmServerError(GvmError):
+    """An exception for gvm server errors
+
+    Base class for all exceptions originating in python-gvm.
```
Comment (bjoernricks) [inline gvm/errors.py]: The class description should be adjusted

File: gvm/errors.py [hunk 2]
```diff
@@ -29,6 +29,46 @@ class GvmError(Exception):
...
+    def __str__(self):
+        return (
+            self.__class__.__name__ + ': ' + self.status + ' - ' + self.message
+        )
+
+
+class GvmClientError(GvmError):
+    """An exception for gvm client errors
+
+    Base class for all exceptions originating in python-gvm.
```
Comment (bjoernricks) [inline gvm/errors.py]: The class description should be adjusted

File: gvm/errors.py [hunk 3]
```diff
@@ -66,6 +106,29 @@ def __str__(self):
         return "Invalid argument {} for {}".format(self.argument, self.function)
 
 
+class InvalidArgumentType(GvmError):
+    def __init__(
```
Comment (bjoernricks) [inline gvm/errors.py]: The error class should get an docstring

**Comment by y0urself** [comment]
> Wow a lot of changes. Looks very good! I just would like to get some small docstring updates for the error classes and gvm/errors.py needs some reformatting with black to satisfy the linter rules.

Yeah. I will add some test cases for `errors.py` and `transforms.py`, too, so the CodeCoverageBot will be happy again ... :)

**Pull request merged**